src/moudule/crud.js

```diff
@@ -1,46 +1,144 @@
-    if (checkbox.checked) {
-      label.style.textDecoration = 'line-through';
-    } else {
-      label.style.textDecoration = 'none';
-    }
-  });
-
-  checkboxContainer.append(checkbox, label);
-  itemTaskElem.append(checkboxContainer, iconTask);
-  li.appendChild(itemTaskElem);
-  return li;
-}
-
-export default Crud;
\ No newline at end of file
+const containerElement = document.querySelector('.containerTodo');
+const ulElement = document.querySelector('.ulElement');
+
+const Task = class {
+  constructor(description, completed = false, index = 1) {
+    this.description = description;
+    this.completed = completed;
+    this.index = index;
+    this.tasks = [
+      {
+        description: 'wash the dogs',
+        completed: false,
+        index: 0,
+      },
+      {
+        description: 'Complete To Do list project',
+        completed: false,
+        index: 1,
+      },
+    ];
+  }
+
+  populateField = () => {
+    let i = -1;
+    this.tasks.forEach((task) => {
+      i += 1;
+      task.index = i;
+    });
+    localStorage.setItem('savedTasks', JSON.stringify(this.tasks));
+  };
+
+  addTask = (newTask) => {
+    this.tasks.push(newTask);
+    this.populateField();
+    this.showTasks();
+  };
+
+  removeTask(task) {
+    const result = this.tasks.filter((b) => b !== task);
+    this.tasks = result;
+    this.populateField();
+  }
+
+  showTasks = () => {
+    ulElement.innerHTML = '';
+    this.tasks.map((task) => {
+      const li = document.createElement('li');
+      const itemElement = document.createElement('div');
+      const checkbox = document.createElement('input');
+      const checkboxContainer = document.createElement('div');
+      checkboxContainer.classList.add('checkboxContainer');
+      const label = document.createElement('label');
+      const icon = document.createElement('div');
+      const textInput = document.createElement('input');
+      textInput.classList.add('textInput', 'hidden');
+      icon.classList.add('iconContainer');
+      itemElement.classList.add('itemElement');
+      icon.innerHTML = '<i class="fas fa-ellipsis-v"></i>';
+      const iconMenu = '<i class="fas fa-ellipsis-v"></i>';
+      const iconDelete = '<i class="fas fa-trash-alt"></i>';
+      textInput.value = task.description;
+      checkbox.type = 'checkbox';
+      checkbox.id = `check${task.index}`;
+      label.innerHTML = task.description;
+      checkbox.checked = task.completed;
+
+      if (checkbox.checked) {
+        label.style.textDecoration = 'line-through';
+      }
+
+      label.addEventListener('click', () => {
+        label.classList.add('hidden');
+        textInput.classList.remove('hidden');
+        li.classList.add('editing');
+        icon.innerHTML = iconDelete;
+        icon.style.cursor = 'pointer';
+        textInput.focus();
+      });
+
+      textInput.addEventListener('focus', () => {
+        this.populateField();
+        icon.innerHTML = iconDelete;
```

Comment:
I suggest you have this delete icon not connected to the edit button. The remove capability is only activated once the item is clicked for editing. One should be able to delete an item without needing to edit it. It provides better user experience.